crates/aptos/src/move_tool/mod.rs

```diff
@@ -594,6 +598,89 @@ impl CliCommand<TransactionSummary> for PublishPackage {
+    pub(crate) account: AccountAddress,
+
+    #[clap(long)]
+    pub(crate) seed: String,
+
+    #[clap(long)]
+    pub(crate) named_address_name: String,
+
+    /// Whether to override the check for maximal size of published data
+    #[clap(long)]
+    pub(crate) override_size_check: bool,
+
+    #[clap(flatten)]
+    pub(crate) included_artifacts_args: IncludedArtifactsArgs,
+    #[clap(flatten)]
+    pub(crate) move_options: MovePackageDir,
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+}
+
+#[async_trait]
+impl CliCommand<TransactionSummary> for CreateResourceAccountAndPublishPackage {
+    fn command_name(&self) -> &'static str {
+        "ResourceAccountPublishPackage"
+    }
+
+    async fn execute(self) -> CliTypedResult<TransactionSummary> {
+        let CreateResourceAccountAndPublishPackage {
+            account,
+            seed,
+            named_address_name: named_address,
+            mut move_options,
+            txn_options,
+            override_size_check,
+            included_artifacts_args,
+        } = self;
+
+        let resource_address =
+            create_resource_address(account, &bcs::to_bytes(&seed.clone()).unwrap());
+        move_options.add_named_address(named_address, resource_address.to_string());
+
+        let package_path = move_options.get_package_path()?;
+        let options = included_artifacts_args
+            .included_artifacts
+            .build_options(move_options.named_addresses());
+        let package = BuiltPackage::build(package_path, options)?;
+        let compiled_units = package.extract_code();
+
+        // Send the compiled module and metadata using the code::publish_package_txn.
+        let metadata = package.extract_metadata()?;
+
+        let message = format!(
+            "Do you want to publish this package under the resource account's address {}?",
+            resource_address
+        );
+        prompt_yes_with_override(&message, txn_options.prompt_options)?;
+
+        let payload = cached_packages::aptos_stdlib::resource_account_create_resource_account_and_publish_package(
+            bcs::to_bytes(&seed)?,
+            bcs::to_bytes(&metadata).expect("PackageMetadata has BCS"),
+            compiled_units,
+        );
+        let size = bcs::serialized_size(&payload)?;
+        println!("package size {} bytes", size);
+        if !override_size_check && size > MAX_PUBLISH_PACKAGE_SIZE {
+            return Err(CliError::UnexpectedError(format!(
+                "The package is larger than {} bytes ({} bytes)! To lower the size \
+                you may want to include less artifacts via `--included_artifacts`. \
+                You can also override this check with `--override-size-check",
+                MAX_PUBLISH_PACKAGE_SIZE, size
+            )));
+        }
+        txn_options
+            .submit_transaction(payload)
+            .await
+            .map(TransactionSummary::from)
+    }
```

Comment:
Similar to `TransactionSummary`, consolidate?

Reply:
could you elaborate?